### PR TITLE
Automate push to maven central repository for releases

### LINKED ISSUE #153: Explore automating maven releases
Right now, releases to maven are done manually, and it would be nice to do this within a script triggered by Travis.  This may be a bit tricky because deploying requires signing the bits.

### PR DESCRIPTION
**Pull request by ajkannan**
Fixes #153 
Associated PR to add the keys is in #204 

cc/ @mziccard: your post on automating releases (from a while back) was really helpful - thank you!  Do these changes look good to you?

### CODE REVIEW

**Comment by ajkannan** [comment]
(Quick update: fixing some build issues, check back in a couple minutes)

**Comment by ajkannan** [comment]
Actually, getting Travis to pass before merging in the commits may not be possible.  Because builds for PRs are considered "untrusted," the secure environment variables needed to decrypt the keys are not exported.  (http://docs.travis-ci.com/user/environment-variables/#Encrypted-Variables).  I think when these changes are merged in, the build should become trusted.

@mziccard I tested the changes on my own private repository a little while ago.  Besides that, do you think there's a better way to test this PR?  (Also feel free to start reviewing when you get the chance, since the build issues don't seem to be caused by the code changes.)

**Comment by mziccard** [comment]
@ajkannan I can't wait to see this up and running:) Anyway I don't think there's a better way to test this other than trying the workflow on your repository (I did something similar today to test key encryption/decryption and environment variables setup for storage integration tests).

File: .travis.yml [hunk 1]
```diff
@@ -6,6 +6,10 @@ jdk:
 before_install:
 - mvn clean
 - git clone -b travis `git config --get remote.origin.url` target/travis
+- openssl aes-256-cbc -K $encrypted_631490ecae8f_key -iv $encrypted_631490ecae8f_iv -in target/travis/signing-tools.tar.enc -out target/travis/signing-tools.tar -d
+- mkdir target/travis/signing-tools
+- chmod 700 target/travis/signing-tools
```
Comment (mziccard) [inline .travis.yml]: Out of curiosity, is `chmod` really needed?

File: utilities/after_success.sh [hunk 2]
```diff
@@ -31,6 +31,8 @@ if [ "${TRAVIS_JDK_VERSION}" == "oraclejdk7" -a "${TRAVIS_BRANCH}" == "master" -
         # Update versions README and pom.xml in master branch
         cd ..
         utilities/update_docs_version.sh
+
```
Comment (mziccard) [inline utilities/after_success.sh]: (nit) why the empty line here?

File: utilities/after_success.sh [hunk 3]
```diff
@@ -31,6 +31,8 @@ if [ "${TRAVIS_JDK_VERSION}" == "oraclejdk7" -a "${TRAVIS_BRANCH}" == "master" -
         # Update versions README and pom.xml in master branch
```
Comment (mziccard) [inline utilities/after_success.sh]: I thing here we are only updating the READMEs, can you confirm that? If so would you mind updating the comment?

File: .travis.yml (hunk 1 above)
Comment (ajkannan) [inline .travis.yml]: I added that line after `gpg` complained that the folder containing the keys was readable by more than the user.

File: utilities/after_success.sh (hunk 2 above)
Comment (ajkannan) [inline utilities/after_success.sh]: I didn't want to associate `mvn deploy` with the comment at the top of the preceding block (update readme versions), but I can make that comment an inline comment and remove this extra line.  I don't feel strongly about the line either way.

File: utilities/after_success.sh (hunk 3 above)
Comment (ajkannan) [inline utilities/after_success.sh]: True, I will update that

**Comment by mziccard** [comment]
Looks good to me. @aozarov do you want to merge this? So we can see if everything is fine with travis:)

**Comment by aozarov** [comment]
@mziccard done, but I think you should be able to merge as well.

**Pull request merged**